extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLExecutionHandler.java

```diff
@@ -123,21 +148,37 @@ private String doRequest(final byte[] body) {
         try (ByteArrayInputStream input = new ByteArrayInputStream(body);
                 final JsonReader jsonReader = Json.createReader(input)) {
             JsonObject jsonInput = jsonReader.readObject();
-            JsonObject outputJson = getExecutionService().execute(jsonInput);
-            if (outputJson != null) {
-                try (StringWriter output = new StringWriter();
-                        final JsonWriter jsonWriter = Json.createWriter(output)) {
-                    jsonWriter.writeObject(outputJson);
-                    output.flush();
-                    return output.toString();
-                }
-            }
-            return null;
+            return doRequest(jsonInput);
         } catch (IOException ex) {
             throw new RuntimeException(ex);
         }
     }
 
+    private String doRequest(JsonObject jsonInput) {
+        JsonObject outputJson = getExecutionService().execute(jsonInput);
+        if (outputJson != null) {
+            try (StringWriter output = new StringWriter();
+                    final JsonWriter jsonWriter = Json.createWriter(output)) {
```

Comment:
Using `Json.*` methods is quite expensive, could we improve performance by caching the result of `JsonProvider.provider()` and using that instead?

Reply:
Done. Thanks :)